lib/veri_traci.ml

```diff
@@ -0,0 +1,79 @@
+open Bap.Std
+open Bap_traces.Std
+open Monad.State
+
+type 'a u = 'a Bil.Result.u
+type event = Trace.event
+
+let stub = fun _ -> return () 
+
+let mem_of_arch arch = 
+  let (module T : Target) = target_of_arch arch in
+  T.CPU.mem
+
+class context trace =
+  object(self:'s)
+    inherit Bili.context
+  end
+
+let mem_operands mv = 
+  let width = Word.bitwidth (Move.data mv) in
+  Size.of_int_opt width |> function
+  | None -> 
+    let er = Printf.sprintf "wrong size for memory operand: %d" width in
+    failwith er
+  | Some sz ->
+    let addr = Bil.int (Move.cell mv) in
+    let data = Bil.int (Move.data mv) in
+    addr, data, sz
+
+class ['a] t arch = 
+  let mem = mem_of_arch arch in 
+  let mem' = Bil.var mem in
```

Comment:
it is better to have `let mem_var = mem_of_arch arch in let mem = Bil.var mem_var in ...` so that you can use `~mem` where `mem` is expected.